### ISSUE TITLE
Saving and loading models does not work as expected.

### ISSUE THREAD

**Issue by miguelroboso** [deepchem/deepchem#870]
By default, models are saved in a temporary directory. 

I tried to change the dir, by changing `model.model_dir` to a new directory where I copied the content of the tmp dir, then saved the model, and loaded it from the new dir again, but Tensorflow notifies me that it loaded the model from the old tmp dir anyway.

```
[testInput.csv.zip](https://github.com/deepchem/deepchem/files/1363610/testInput.csv.zip)
[testInput.csv.zip](https://github.com/deepchem/deepchem/files/1363614/testInput.csv.zip)

from __future__ import division
from __future__ import print_function
from __future__ import unicode_literals

import numpy as np
import tensorflow as tf
import deepchem as dc
from deepchem.models.tensorgraph.models.graph_models import GraphConvTensorGraph


BATCH_SIZE = 128

from deepchem.utils.save import load_from_disk

dataset_file= "testInput.csv"
dataset = load_from_disk(dataset_file)
print("Columns of dataset: %s" % str(dataset.columns.values))
print("Number of examples in dataset: %s" % str(dataset.shape[0]))

import deepchem as dc

# featurizer = dc.feat.CircularFingerprint(size=1024)
featurizer = dc.feat.ConvMolFeaturizer()

loader = dc.data.CSVLoader(
      tasks=["active"], smiles_field="SMILES",
      featurizer=featurizer)
dataset = loader.featurize(dataset_file)

splitter = dc.splits.ScaffoldSplitter(dataset_file)
train_dataset, valid_dataset, test_dataset = splitter.train_valid_test_split(
    dataset, seed = 1138)

transformers = [dc.trans.NormalizationTransformer(transform_y=True, dataset=train_dataset)]

for dataset in [train_dataset, valid_dataset, test_dataset]:
  for transformer in transformers:
      dataset = transformer.transform(dataset)

model = GraphConvTensorGraph(
    1, batch_size=128, mode='classification')
model.fit(train_dataset, nb_epoch=1)


# check model_dir

model.model_dir

# copy from tmp dir

import shutil
model.save()
shutil.copytree(model.model_dir, './testmodel')

# check that it's copied

import os
os.listdir('.')

# load

model = GraphConvTensorGraph.load_from_dir('./testmodel')

# the dir is the original one

model.model_dir

# setting model_dir before saving

model.model_dir = os.path.abspath('./testmodel')

# check that the right dir is there

model.model_dir

# saving after changing model_dir

model.save()

# trying to load again, TF will tell me that it loaded params from the tmp dir

model = GraphConvTensorGraph.load_from_dir('./testmodel')


# model_dir is the appropriate one

model.model_dir

```

**Comment by miguelroboso** [comment]
Input file
[testInput.csv.zip](https://github.com/deepchem/deepchem/files/1363618/testInput.csv.zip)

**Comment by lilleswing** [comment]
ah in the tg.load_from_dir we have to override the pickled version of model_dir with the one passed in.

```python
  @staticmethod
  def load_from_dir(model_dir):
    pickle_name = os.path.join(model_dir, "model.pickle")
    with open(pickle_name, 'rb') as fout:
      tensorgraph = pickle.load(fout)
      tensorgraph.built = False
      tensorgraph.model_dir = model_dir #  Add this line
      try:
        tensorgraph.restore()
      except ValueError:
        pass  # No checkpoint to load
      return tensorgraph
```

Pull requests welcome!

**Comment by rbharath** [comment]
Thanks! Closing since should be fixed by #871.
